src/ol/style/regularshape.js

```diff
@@ -401,11 +431,12 @@ ol.style.RegularShape.prototype.draw_ = function(renderOptions, context, x, y) {
   context.translate(x, y);
 
   context.beginPath();
+  var points = this.points_;
   if (this.radius2_ !== this.radius_) {
-    this.points_ = 2 * this.points_;
+    points = 2 * points;
```

Comment:
Ok, i agree with you, it is better to seperate the concerns. I open another issue for this. I thought that about the moving of the file, too. Thanks for the review!